common/spring-cloud-stream-modules-test-support/src/main/java/org/springframework/cloud/stream/modules/test/file/remote/TestSftpServer.java

```diff
@@ -38,20 +41,73 @@
 
 	private final SshServer server = SshServer.setUpDefaultServer();
 
-	private final int port;
-
 	public TestSftpServer(String root, int port) {
 		super(root);
-		this.port = port;		
+		setFtpTemporaryFolder(
+				new TemporaryFolder() {
```

Comment:
I had to do a bunch of rework here in the next PR (FTP sink) so I could reuse it for the sink.

Let's hold off from making more changes here until the next PR.